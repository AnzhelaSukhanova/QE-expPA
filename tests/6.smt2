(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(declare-fun z () (_ BitVec 4))
(declare-fun t () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and 
 (or (bvult x (bvmul #b0011 y)) (= x (bvmul #b0011 y))) 
 (or (bvult z x) (= z x)) 
 (or (bvult x t) (= x t)))))
