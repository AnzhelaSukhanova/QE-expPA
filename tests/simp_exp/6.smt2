(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(declare-fun z () (_ BitVec 4))
(declare-fun t () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and 
 (or (bvult (bvshl #b0001 x) (bvmul #b0111 y)) (= (bvshl #b0001 x) (bvmul #b0111 y))) 
 (or (bvult (bvshl #b0001 x) z) (= (bvshl #b0001 x) z)) 
 (or (bvult (bvshl #b0001 x) (bvshl #b0001 t)) (= (bvshl #b0001 x) (bvshl #b0001 t))))))
