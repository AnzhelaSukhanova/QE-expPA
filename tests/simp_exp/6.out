(let (($e1 (bvmul y #b0111))) (let (($e2 (bvshl #b0001 t))) (let (($e3 (= $e2 #b0000))) (let (($e4 (ite (bvult $e2 #b1000) (ite (bvult $e2 #b0100) (ite (bvult $e2 #b0010) (ite $e3 #b1111 #b0000) #b0001) #b0010) #b0011))) (let (($e5 (bvshl #b0001 $e4))) (let (($e6 (bvshl #b0001 (bvadd #b0001 $e4)))) (let (($e7 (= #b0001 $e2))) (let (($e8 (= $e2 #b0010))) (let (($e9 (= $e2 #b0100))) (let (($e10 (= #b1000 $e2))) (let (($e11 (bvult #b0001 $e2))) (let (($e12 (bvult #b0010 $e2))) (let (($e13 (bvult #b0100 $e2))) (let (($e14 (bvult #b1000 $e2))) (let (($e15 (= $e1 #b0000))) (let (($e16 (ite (bvult $e1 #b1000) (ite (bvult $e1 #b0100) (ite (bvult $e1 #b0010) (ite $e15 #b1111 #b0000) #b0001) #b0010) #b0011))) (let (($e17 (bvshl #b0001 $e16))) (let (($e18 (bvshl #b0001 (bvadd #b0001 $e16)))) (let (($e19 (= #b0001 $e1))) (let (($e20 (= $e1 #b0010))) (let (($e21 (= $e1 #b0100))) (let (($e22 (= #b1000 $e1))) (let (($e23 (bvult #b0001 $e1))) (let (($e24 (bvult #b0010 $e1))) (let (($e25 (bvult #b0100 $e1))) (let (($e26 (bvult #b1000 $e1))) (let (($e27 (= z #b0000))) (let (($e28 (ite (bvult z #b1000) (ite (bvult z #b0100) (ite (bvult z #b0010) (ite $e27 #b1111 #b0000) #b0001) #b0010) #b0011))) (let (($e29 (= z #b0001))) (let (($e30 (= z #b0010))) (let (($e31 (= z #b0100))) (let (($e32 (= z #b1000))) (let (($e33 (bvult #b0001 z))) (let (($e34 (bvult #b0010 z))) (let (($e35 (bvult #b0100 z))) (let (($e36 (bvult #b1000 z))) (and (not (and (not $e24) (not $e23) (not $e20) (not $e19) (not $e25) (not $e21) (not $e26) (not $e22) $e15 (not $e15) (not $e23) (not $e19) (not $e24) (not $e20) (not $e25) (not $e21) (not $e26) (not $e22) $e15 (not $e15) (not $e23) (not $e19) (not (bvult $e18 $e1)) (not (bvult $e17 $e1)) (not $e24) (not (= $e1 $e18)) (not (= $e1 $e17)) (not $e20))) (not (and (not $e12) (not $e11) (not $e8) (not $e7) (not $e13) (not $e9) (not $e14) (not $e10) $e3 (not $e3) (not $e11) (not $e7) (not $e12) (not $e8) (not $e13) (not $e9) (not $e14) (not $e10) $e3 (not $e3) (not $e11) (not $e7) (not (bvult $e6 $e2)) (not (bvult $e5 $e2)) (not $e12) (not (= $e2 $e6)) (not (= $e2 $e5)) (not $e8))) (not (and (not $e34) (not $e33) (not $e35) (not $e30) (not $e29) (not $e36) (not $e31) $e27 (not $e32) (not $e33) (not $e27) (not $e34) (not $e29) (not $e35) (not $e30) (not $e36) (not $e31) $e27 (not $e32) (not $e33) (not $e27) (not (bvult (bvshl #b0001 (bvadd #b0001 $e28)) z)) (not (bvult (bvshl #b0001 $e28) z)) (not $e34) (not $e29) (not $e30))))))))))))))))))))))))))))))))))))))))
